api/admin/controller/__init__.py

```diff
@@ -1562,20 +1577,22 @@ def _get_prior_test_results(self, item, protocol_class=None):
 
         return self_test_results
 
-    def _mirror_integration_setting(self):
+    def _mirror_integration_setting(self, type=""):
         """Create a setting interface for selecting a storage integration to
         be used when mirroring items from a collection.
         """
         integrations = self._db.query(ExternalIntegration).filter(
             ExternalIntegration.goal==ExternalIntegration.STORAGE_GOAL
         ).order_by(
             ExternalIntegration.name
-        ).all()
+        )
+
         if not integrations:
             return
+        key = "%s_mirror_integration_id" % type.lower() if type else "mirror_integration_id"
         mirror_integration_setting = {
-            "key": "mirror_integration_id",
-            "label": _("Mirror"),
+            "key": key,
+            "label": _("%s Mirror" % type),
             "description": _("Any cover images or free books encountered while importing content from this collection can be mirrored to a server you control."),
```

Comment:
The description also needs to be changed to match the type of thing being mirrored. Maybe `label` and `description` should be separate arguments to this method; then the caller would have complete control over how the UI is presented.

Reply:
In fact, it might be smart to put all the UI configuration in ExternalIntegrationLink, since thats' where we keep the constants COVERS and BOOKS.